### PR TITLE
[s3] Remove unnecessary public access to S3 buckets created under test

### PR DESCRIPTION
**Pull request by desokroshan**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/CleanupBucketIntegrationTests.java [hunk 1]
```diff
@@ -61,7 +61,11 @@ public void testCleanup() {
                     || bucket.getName().startsWith("s3-low-level-presigned-url")
                     || bucket.getName().startsWith("java-bucket-policy-integ-test")
                     || bucket.getName().startsWith("copy-object-integ-test")
-                    || bucket.getName().startsWith("java-server-side-encryption-integ-test")) {
+                    || bucket.getName().startsWith("java-server-side-encryption-integ-test")
+                    || bucket.getName().startsWith("acceleration-integration-test-")
+                    || bucket.getName().startsWith("amazon-transfer-util-integ-test")
+                    || bucket.getName().startsWith("java-bucket-lifecycle-integ-test-")
+                    || bucket.getName().startsWith("java-s3-object-iteration-test-")) {
```
Comment (jpignata) [inline aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/CleanupBucketIntegrationTests.java:68]: It's hard to look at this without thinking there's a better way to approach this such as a standard prefix or using S3 tags to indicate it's a test bucket. Thoughts?

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/UnauthenticatedIntegrationTest.java [hunk 2]
```diff
@@ -65,7 +65,6 @@ public void tearDown() {
     @Test
     public void testUnauthenticatedOperations() {
         s3.createBucket(bucketName);
-        s3.setBucketAcl(bucketName, CannedAccessControlList.PublicRead);
```
Comment (jpignata) [inline aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/UnauthenticatedIntegrationTest.java:68]: Is this change sufficient? How will the below line that calls ` PutObjectRequest` succeed? Did you run these tests?

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/V1BucketAddressingIntegrationTest.java [hunk 3]
```diff
@@ -75,7 +75,6 @@ public void testV1BucketAddressing() throws Exception {
         createV1AddressedBucket(bucketName);
         file = super.getRandomTempFile("v1-addressing-integ-test", 333L);
 
-        s3.setBucketAcl(bucketName, CannedAccessControlList.PublicRead);
```
Comment (jpignata) [inline aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/V1BucketAddressingIntegrationTest.java:78]: Same question. What is this testing, and why was this public?

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/UnauthenticatedIntegrationTest.java (hunk 2 above)
Comment (jpignata) [reply aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/UnauthenticatedIntegrationTest.java:68]: So essentially this line was unnecessary and the test still passes without it?

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/V1BucketAddressingIntegrationTest.java (hunk 3 above)
Comment (jpignata) [reply aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/V1BucketAddressingIntegrationTest.java:78]: Same here?

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/UnauthenticatedIntegrationTest.java (hunk 2 above)
Comment (desokroshan) [reply aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/UnauthenticatedIntegrationTest.java:68]: Yeah these seem redundant and the tests still pass without it, so removing would suffice in this case.

**Comment by desokroshan** [comment]
> What's the plan for the other reference(s)?
> 
> For example:
> 
> https://github.com/aws-amplify/aws-sdk-android/blob/881bb946777a56488c659c8ef8c210d1bf689f3d/aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/AclIntegrationTest.java#L136

Here the test explicitly is to check if we can set public read ACL on a bucket and there is a assert statement to this effect. So I don't think there is a way to remove this without actually getting rid of the test case itself.

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/CleanupBucketIntegrationTests.java (hunk 1 above)
Comment (desokroshan) [reply aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/CleanupBucketIntegrationTests.java:68]: We can for some of these but interestingly a lot these patterns are not created by our integration tests. I guess these are being created by tests from other teams (not sure why?). So we might have trace their origin and make a fix accordingly.

**Comment by jpignata** [comment]
> Here the test explicitly is to check if we can set public read ACL on a bucket and there is a assert statement to this effect. So I don't think there is a way to remove this without actually getting rid of the test case itself.

The idea here is to test our code and its ability to set a canned ACL -- not for us to test every canned ACL. Why don't we simply remove this canned ACL and test others that don't potentially introduce security concerns? What are the downsides in terms of the code we're trying to exercise?

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/CleanupBucketIntegrationTests.java (hunk 1 above)
Comment (jpignata) [reply aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/CleanupBucketIntegrationTests.java:68]: We're sharing the account with other processes? If so, let's fix that immediately to make this easier.

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/CleanupBucketIntegrationTests.java [hunk 4]
```diff
@@ -61,7 +61,11 @@ public void testCleanup() {
                     || bucket.getName().startsWith("s3-low-level-presigned-url")
                     || bucket.getName().startsWith("java-bucket-policy-integ-test")
                     || bucket.getName().startsWith("copy-object-integ-test")
-                    || bucket.getName().startsWith("java-server-side-encryption-integ-test")) {
+                    || bucket.getName().startsWith("java-server-side-encryption-integ-test")
+                    || bucket.getName().startsWith("acceleration-integration-test-")
```
Comment (jamesonwilliams) [inline aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/CleanupBucketIntegrationTests.java:65]: The existing code construct here is absolutely batty.

At _minimum_, it should be like:
```
List<String> bucketPrefixes = Arrays.asList(
    "acceleration-integration-test-",
   ...
);
for (String prefix : bucketPrefixes) {
    if (bucket.getName().startsWith(prefix)) {
        // do stuff
    }
}
```

From there, generating the `List<String>` in a more automated way (from file, from inspection of test data, etc.) would be desirable.

@jpignata's suggestion of using an AWS Tag on the bucket like `special-test-slug-55` or whatever would also be a nice improvement.

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/mobileconnectors/s3/transferutility/PauseTransferIntegrationTest.java [hunk 5]
```diff
@@ -32,7 +32,7 @@
 
 public final class PauseTransferIntegrationTest extends S3IntegrationTestBase {
 
-    private static final String BUCKET_NAME = "amazon-transfer-util-integ-test-" + new Date().getTime();
+    private static final String BUCKET_NAME = "android-sdk-transfer-util-integ-test-" + new Date().getTime();
```
Comment (jpignata) [inline aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/mobileconnectors/s3/transferutility/PauseTransferIntegrationTest.java:35]: In some of these constructions, we're using `Date().getTime()`, in others: `System.currentTimeMillis()` -- what do you think about being consistent?

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/CleanupBucketIntegrationTests.java (hunk 4 above)
Comment (desokroshan) [reply aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/CleanupBucketIntegrationTests.java]: Updated to use a common prefix for all the test buckets.

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/AclIntegrationTest.java [hunk 6]
```diff
@@ -144,8 +143,6 @@ public void testCannedAcls() throws Exception {
         for (int index = 0; index < accessControls.length; index++) {
```
Comment (jpignata) [inline aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/AclIntegrationTest.java:143]: What is this test enforcing now? That only assert is checking bucket owner, which seems far removed from its original intent.

File: aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/AclIntegrationTest.java [hunk 7]
```diff
@@ -124,35 +125,35 @@ public void testCreateBucketWithAcl() throws Exception {
...
         // Public Read Canned ACL
-        s3.setBucketAcl(bucketName, CannedAccessControlList.PublicRead);
-        s3.setObjectAcl(bucketName, key, CannedAccessControlList.PublicRead);
-        AccessControlList bucketAcl = s3.getBucketAcl(bucketName);
-        AccessControlList objectAcl = s3.getObjectAcl(bucketName, key);
+        s3.setObjectAcl(BUCKET_NAME, KEY, CannedAccessControlList.PublicRead);
+        AccessControlList bucketAcl = s3.getBucketAcl(BUCKET_NAME);
+        AccessControlList objectAcl = s3.getObjectAcl(BUCKET_NAME, KEY);
 
         AccessControlList[] accessControls = new AccessControlList[] {
```
Comment (jpignata) [inline aws-android-sdk-s3-test/src/androidTest/java/com/amazonaws/services/s3/AclIntegrationTest.java:141]: What is this used for? I see the variable being overridden below, is this necessary, or should we just move the declration to line 157?

**Pull request merged**